test/classes/sequelize-service.test.ts

```diff
@@ -210,14 +209,28 @@ describe('SequelizeService', function () {
       expect(user).to.have.keys(['email', 'id']); // ID is auto selected
     });
 
+    it('should only resolve types for what was selected (+ auto selected id)', async () => {
+      await userService.create({ email: 'foo', password: 'bar' });
+      const [ user ] = await userService.find({}, { select: ['email'] });
+      expect(user).to.have.keys(['email', 'id']); // ID is auto selected
+      expect(user).to.not.have.keys(['first_name']);
+      // Note: user.first_name should not even compile!
```

Comment:
@SylvainEstevez `user.isAdult` compiles. That's not intended, right?

Reply:
Definitely not, since it’s not in compute

Reply:
(Note: it doesn't have that actual key, typescript just thinks so)

Yeah I think there is a "lucky default" going on somewhere. Will fix...

Reply:
`compute` in followup PR: https://github.com/bluebirds-blue-jay/sequelize-service/pull/5